packages/amplication-git-utils/src/git/git.service.ts

```diff
@@ -89,39 +99,213 @@ export class GitClientService {
     }
 
     if (pullRequestMode === EnumPullRequestMode.Accumulative) {
-      await this.provider.createBranchIfNotExists({
+      const gitClient = new GitClient();
+      const cloneFolder = process.env.CLONES_FOLDER;
+      if (!cloneFolder) {
+        throw new MissingEnvParam("CLONES_FOLDER");
+      }
+
+      const randomUUID = v4();
+
+      const cloneUrl = this.provider.getCloneUrl({
+        owner,
+        repositoryName,
+      });
+
+      const cloneDir = normalize(
+        join(cloneFolder, this.provider.name, owner, repositoryName, randomUUID)
+      );
+
+      await gitClient.clone(cloneUrl, cloneDir);
+
+      await this.restoreAmplicationBranchIfNotExists({
         owner,
         repositoryName,
         branchName,
-      }),
-        await this.provider.createCommit({
+        gitClient,
+      });
+
+      const diffFolder = normalize(
+        join(
+          `.amplication/diffs`,
+          this.provider.name,
           owner,
+
           repositoryName,
-          commitMessage,
-          branchName,
-          files: preparedFiles,
+          randomUUID
+        )
+      );
+
+      const { diff } = await this.preCommitProcess({
+        branchName,
+        gitClient,
+        owner,
+        repositoryName,
+      });
+      let fullDiffPath = "";
+      if (diff) {
+        await mkdir(diffFolder, { recursive: true });
+        const diffPath = join(diffFolder, "diff.patch");
+        await writeFile(diffPath, diff);
+        fullDiffPath = resolve(diffPath);
+        console.log("Saving diff to: ", fullDiffPath);
+      }
+
+      await this.provider.createCommit({
+        owner,
+        repositoryName,
+        commitMessage,
+        branchName,
+        files: preparedFiles,
+      });
+
+      if (diff) {
+        await this.postCommitProcess({
+          diffPath: fullDiffPath,
+          gitClient,
         });
+        await rm(fullDiffPath);
+      }
+
+      await rm(cloneDir, { recursive: true, force: true });
+
       const { defaultBranch } = await this.provider.getRepository({
         owner,
         repositoryName,
       });
+
       const existingPullRequest = await this.provider.getPullRequestForBranch({
         owner,
         repositoryName,
```

Comment:
Let's avoid setting `undefined` as expected values. We should try to use `null` (it will affect the interface too)